tests/data/test_metapath_walker.py

```diff
@@ -316,13 +317,17 @@ def test_init_parameters(self):
         assert all(np.array_equal(w1, w2) for w1, w2 in zip(run_1, run_2))
 
     def test_benchmark_uniformrandommetapathwalk(self, benchmark):
-
-        g = create_test_graph()
+        g = example_graph_random(n_nodes=50, n_edges=500, node_types=2)
         mrw = UniformRandomMetaPathWalk(g)
 
-        nodes = ["0"]
+        # this should be made larger to be more realistic, when it is fast enough
+        nodes = np.arange(0, 5)
         n = 5
         length = 5
-        metapaths = [["s", "n", "n", "s"], ["n", "s", "n"], ["n", "n"]]
+        metapaths = [
+            ["n-0", "n-1", "n-1", "n-0"],
+            ["n-0", "n-1", "n-0"],
+            ["n-0", "n-0"],
```

Comment:
Does `example_graph_random` guarantee that edge types do exist that traverse the graph this way, or is it relying on `n_edges` to be large enough that the chance is very low that we miss one? I think otherwise the random walks end up stopping early

Reply:
Hm, you're correct that it doesn't guarantee it, but I think it's rather likely across, with 500 total edges and 100 nodes across 2 node types.

Of course, this has to be qualified by only having 5 head nodes. The average degree of each node in the graph is 10, so there's ~50 edges touching the 5 starting `n-0` starting nodes.

Using the fact that the four possibilities of source/target types for each edge are all equally likely (`1/4`), I think the probabilities of having none of the three paths is low:

- `n-0`--`n-0`: is the chance of every edge having source or target `n-1`, which is `(3/4)**50` &approx; 5.7e-7.
- `n-0 -- n-1 -- n-0`: is the chance of no `n-0 -- n-1` edges (since if there's one, it can follow that edge forwards and backwards), which `(1/2)**50` &approx; 8.9e-16
- `n-0 -- n-1 -- n-1 -- n-0`: is likely going to be larger and seems more complicated to calculate (what's the chance of every `n-0 -- n-1` going to an `n-1` with no edges to other `n-1`), I guess I can work it out, but I don't think it's too important because it's probably also small?

One approach to be more sure would be increasing the `np.arange(0, 5)` to `np.arange(0, 10)` or something?

Thoughts?

Reply:
The numbers sound small enough it's probably fine to leave as is 👍 thanks for checking!